app/src/main/java/mollie/tictactoe/StateConverter.java

```diff
@@ -0,0 +1,66 @@
+package mollie.tictactoe;
+
+import mollie.tictactoe.players.MobilePlayer;
+import ttt.Player;
+import ttt.game.Board;
+import ttt.game.GameEngine;
+import ttt.game.Marks;
+
+public class StateConverter {
```

Comment:
@felipesere I think I understand what you're saying here, but am currently unsure of how to solve it.  The point of this class is really to convert custom objects/enums to strings so that they can be saved to the `savedInstanceState` Bundle, or convert the Strings that have been read from the Bundle back to custom objects.

I could push the conversion back a layer to the StateManager which is what is reading and writing to the Bundle, but I felt that would give too much responsibility to the class , and make it harder to test as the StateManager class is dealing with Android components and not just plain Java.